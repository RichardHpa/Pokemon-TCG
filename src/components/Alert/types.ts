import type { ReactNode } from 'react'

export interface AlertProps {
  status?: 'info' | 'success' | 'error' | 'warning' | 'dark'
  children: ReactNode
}
