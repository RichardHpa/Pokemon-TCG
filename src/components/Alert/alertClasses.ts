export const alertClasses = {
  base: 'flex items-center p-4 mb-4 text-sm border rounded-lg',
  status: {
    info: 'text-blue-800 border-blue-300 bg-blue-50 dark:bg-gray-800 dark:text-blue-400 dark:border-blue-800',
    success:
      'text-green-800 border-green-300 dark:text-green-400 dark:border-green-800 bg-green-50 dark:bg-gray-800',
    error:
      'text-red-800 border-red-300 bg-red-50 dark:bg-gray-800 dark:text-red-400 dark:border-red-800',
    warning:
      'text-yellow-800 border-yellow-300 bg-yellow-50 dark:bg-gray-800 dark:text-yellow-300 dark:border-yellow-800',
    dark: 'text-gray-800 border-gray-300 bg-gray-50 dark:bg-gray-800 dark:text-gray-300 dark:border-gray-600',
  },
}
