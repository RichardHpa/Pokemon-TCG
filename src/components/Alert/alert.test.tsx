import { render, screen } from '@testing-library/react'

import { Alert } from './Alert'

describe('Alert', () => {
  test('renders an Alert', () => {
    const { container } = render(<Alert>alert children</Alert>)
    expect(screen.getByRole('alert')).toBeInTheDocument()

    expect(container).toMatchSnapshot()
  })

  test.each([
    ['info', 'border-blue-300 '],
    ['success', 'border-green-300'],
    ['error', 'border-red-300'],
    ['warning', 'border-yellow-300'],
    ['dark', 'border-gray-300'],
  ] as const)('renders Alert with status:%s', (status, expected) => {
    render(<Alert status={status}>Test</Alert>)
    expect(screen.getByRole('alert')).toHaveClass(expected)
  })
})
