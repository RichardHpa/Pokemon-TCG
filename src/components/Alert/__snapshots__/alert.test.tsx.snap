// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert renders an Alert 1`] = `
<div>
  <div
    class="flex items-center p-4 mb-4 text-sm border rounded-lg text-blue-800 border-blue-300 bg-blue-50 dark:bg-gray-800 dark:text-blue-400 dark:border-blue-800"
    role="alert"
  >
    <svg
      aria-hidden="true"
      class="flex-shrink-0 inline w-4 h-4 mr-3"
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10 .5a9.5 9.5 0 1 0 9.5 9.5A9.51 9.51 0 0 0 10 .5ZM9.5 4a1.5 1.5 0 1 1 0 3 1.5 1.5 0 0 1 0-3ZM12 15H8a1 1 0 0 1 0-2h1v-3H8a1 1 0 0 1 0-2h2a1 1 0 0 1 1 1v4h1a1 1 0 0 1 0 2Z"
      />
    </svg>
    <span
      class="sr-only"
    >
      Info
    </span>
    <div>
      alert children
    </div>
  </div>
</div>
`;
